paramiko/transport.py

```diff
@@ -1125,7 +1125,7 @@ def connect(
         # check host key if we were given one
         # If GSS-API Key Exchange was performed, we are not required to check
         # the host key.
-        if (hostkey is not None) and not gss_kex:
+        if (hostkey is not None) and not self.gss_kex_used:
```

Comment:
is this "too early"? might `gss_kex_used` be set just below via `self.auth_gssapi_keyex(username)`?

Reply:
``gss_kex_used`` is not set in ``transport.py``. It is set in ``kex_gss.py``. The host key check is performed during key exchange (or not, if GSS-API is used) and therefore must be done in ``Transport.connect()``.

In ``Transport.auth_gssapi_keyex(username)`` the user is authenticated if a successful GSS-API key exchange was performed. That's a different layer in the SSH protocol.

Reply:
I see, makes sense (gss is setup via `start_client()` -> `start()` -> `run()` -> `_send_kex_init()` / `_expect_packet()`) - I'm not familiar with gss, just clarifying :)